### PR TITLE
wireless/bluetooth: add interrupt_context hander for netsnoop

### PR DESCRIPTION
**Pull request by chengkai15**
Signed-off-by: chengkai <chengkai@xiaomi.com>

## Summary
add bt bridge snoop 

## Impact
none

## Testing
pass

### CODE REVIEW

File: net/utils/net_snoop.c [hunk 1]
```diff
@@ -161,76 +161,128 @@ begin_packed_struct struct snoop_packet_header_s
...
+ *
+ ****************************************************************************/
+
+static int snoop_flush_lock(FAR struct snoop_s *snoop)
+{
+  irqstate_t flags;
+  int ret;
+
+  flags = enter_critical_section();
+  nxmutex_lock(&snoop->mutex);
```
Comment (xiaoxiang781216) [inline net/utils/net_snoop.c:267]: why use both critical section and nxmutex

File: net/utils/net_snoop.c [hunk 2]
```diff
@@ -161,76 +161,128 @@ begin_packed_struct struct snoop_packet_header_s
...
+}
 
-  ret = file_write(&snoop->filep, &header, sizeof(header));
-  if (ret != sizeof(header))
+/****************************************************************************
+ * Name: snoop_flush
+ *
+ * Description:
+ *   This function could flush snoop buf into file.
+ *
```
Comment (acassis) [inline net/utils/net_snoop.c:218]: Please include Parameter and Return info

File: net/utils/net_snoop.c [hunk 3]
```diff
@@ -161,76 +161,128 @@ begin_packed_struct struct snoop_packet_header_s
...
+
+  return ret;
+}
+
+/****************************************************************************
+ * Name: snoop_flush_work
+ *
+ * Description:
+ *   Do snoop flush work.
+ *
```
Comment (acassis) [inline net/utils/net_snoop.c:280]: Ditto

File: net/utils/net_snoop.c (hunk 1 above)
Comment (chengkai15) [reply net/utils/net_snoop.c:267]: snoop_flush could be called by interrupt context or app context. 
when in interrupt context case, snoop_flush involves with file sync and write operations，which could switch from interrupt context to app context, causing data race. 
so here need enter_critical_section and nxmutex_lock double check

File: net/utils/net_snoop.c (hunk 3 above)
Comment (chengkai15) [reply net/utils/net_snoop.c:280]: done

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**